src/backends/plonky2/basetypes.rs

```diff
@@ -2,6 +2,10 @@
 //! `backend_plonky2` feature is enabled.
 //! See src/middleware/basetypes.rs for more details.
 
+use crate::frontend::serialization::{
+    deserialize_hash_tuple, deserialize_value_tuple, serialize_hash_tuple, serialize_value_tuple,
+};
```

Comment:
I would propose moving these serialize and deserialize functions to the middleware.

Reply:
Done!